usdt-impl/src/common.rs

```diff
@@ -42,7 +42,14 @@ pub fn generate_type_check(types: &[dtrace_parser::DataType]) -> TokenStream {
 // registers.
 pub fn construct_probe_args(types: &[dtrace_parser::DataType]) -> (TokenStream, TokenStream) {
     // TODO this will fail with more than 6 parameters.
-    let abi_regs = ["rdi", "rsi", "rdx", "rcx", "r8", "r9"];
+    let abi_regs = if cfg!(target_arch = "x86_64") {
+        // x86 is first as that provides the stricter limit on the number of arguments
+        ["rdi", "rsi", "rdx", "rcx", "r8", "r9"]
+    } else if cfg!(any(target_arch = "arm64", target_arch = "aarch64")) {
+        ["x0", "x1", "x2", "x3", "x4", "x5"]
+    } else {
+        compile_error!("USDT only supports x86_64 and ARM64 architectures");
```

Comment:
This line is always going to be tripped at compile time. You need to use `#[cfg(..)]` attributes instead as those will actually work as expected with the conditional compilation here.

Reply:
Something like:
```Rust
    #[cfg(target_arch = "x86_64")]
    const ABI_REGS: &[&str] = &["rdi", "rsi", "rdx", "rcx", "r8", "r9"];

    #[cfg(target_arch = "aarch64")]
    const ABI_REGS: &[&str] = &["x0", "x1", "x2", "x3", "x4", "x5"];

    #[cfg(not(any(target_arch = "x86_64", target_arch = "aarch64")))]
    compile_error!("USDT only supports x86_64 and ARM64 architectures");
```

Reply:
Thanks @luqmana I see this, now:

> cfg!, unlike #[cfg], does not remove any code and only evaluates to true or false. For example, all blocks in an if/else expression need to be valid when cfg! is used for the condition, regardless of what cfg! is evaluating.